File: samples/three-layer-kmp-domain-simple/desktopApp/src/main/kotlin/com/example/kmpsamples/threelayerkmpdomainsimple/desktop/HomeViewModel.kt
package com.example.kmpsamples.threelayerkmpdomainsimple.desktop

import com.example.kmpsamples.threelayerkmpdomainsimple.shareddomain.GetTasksUseCase

class HomeViewModel(
    private val useCase: GetTasksUseCase = GetTasksUseCase(TaskRepository(TaskDataSource()))
) {
    fun screenState(platform: String): String {
        val task = useCase.execute(platform)
        return "Native HomeViewModel -> "+ task.title + "\n" + task.details
    }

    fun refresh(platform: String): String {
        val task = useCase.execute(platform)
        return "Native HomeViewModel refreshed -> "+ task.title + "\n" + task.details
    }
}

File: samples/16_three-layer-kmp-domain/iosApp/Sources/HomeViewModel.swift
import SharedDomain

final class HomeViewModel {
    private let useCase: GetTasksUseCase

    init(useCase: GetTasksUseCase = GetTasksUseCase(repository: IosTaskRepository(dataSource: TaskDataSource()))) {
        self.useCase = useCase
    }

    func screenState(platform: String) -> String {
        let task = useCase.execute(platform: platform)
        return "Native HomeViewModel -> \(task.title)\n\(task.details)"
    }

    func refresh(platform: String) -> String {
        let task = useCase.execute(platform: platform)
        return "Native HomeViewModel refreshed -> \(task.title)\n\(task.details)"
    }
}
